docs/rfc/0000-application-architecture.md

```diff
@@ -0,0 +1,506 @@
+    response-headers,
+    ? payload: bstr,
+]
+
+error = [
+    response-headers,
+    code: uint,
+    ? message: tstr,
+]
+
+request-headers = {
+    ua: client-id,
+    rq: request-id,
+    ? token: token,
+}
+
+response-headers = {
+    rq: request-id,
+}
+
+; Unambiguous, human-readable string identifying the client application. Mainly
+; for diagnostic purposes. Example: "radicle-link-cli/v1.2+deaf"
+client-id: tstr .size (4..16)
+
+; Request identifier, choosen by the client. The responder includes the
+; client-supplied value in the response, enabling request pipelining.
+;
+; Note that streaming / multi-valued responses may include the same id in
+; several response messages.
+request-id: bstr .size (4..16)
+
+; Placeholder for future one-time-token support.
+token: bstr
+```
+
+
+## Architecture
+
+With the prerequisites in place, we lay out the following architecture:
+
+```
+                                   +-----------------+
+                                   |                 |
+         +-------------------------+    Native App   +------------------------+
+         |                         |                 |                        |
+         |                         +--------+--------+                        |
+         |                                  |                                 |
+         |                      +-----------+----------+                      |
+         |                      |                      |                      |
+         |                      |                      |                      |
++--------+--------+    +--------+--------+    +--------+--------+    +--------+--------+   +----------------+
+|                 |    |                 |    |                 |    |                 |   |                |
+|  peer-to-peer   |    |     CLIaaS      |    |    ssh-agent    |    |      PubSub     +---+      gitd      |
+|                 |    |                 |    |                 |    |                 |   |                |
++--------+--------+    +-----------------+    +--------+--------+    +--------+--------+   +-------+--------+
+         |                                             |         \            |           /        |
+         |                                 +-----------+          +----------------------+         |
+         |                                 |                                  |           \        |
+         |                         +-------+--------+                         |            +-------+--------+
+         |                         |                |                         |            |                |
+         +-------------------------+      CLI       +-------------------------+            |      git       |
+                                   |                |                                      |                |
+                                   +----------------+                                      +----------------+
+```
```

Comment:
It would be super helpful to have a brief description of each box and what they suppose to do. Trying to understand each component and the reason for isolation.

Reply:
The components are described in the paragraphs following this masterpiece -- can I make this clearer in some way?

Reply:
I would suggest

- Using the exact same name in the headlines, without renaming or enhancing the text.
- Have a simple list before the diagram where people could read the name (pointing to the bigger content that follow) and a really short description
- Include all the component names some where (even if the description is short)

Take my feedback with a grain of salt, this is my perspective as somebody that has less than 2 hours of reading stuff (so newbie perspective)